helixbass/tsc-rust
# Bind variable declaration to symbol

In this PR:
- create symbol table and populate with a symbol corresponding to a variable declaration during binding

To test:
If you have a file named eg `tmp.tsx` in the repo root directory whose contents are `const a: number = true` and run `cargo run tmp.tsx` you should see debug-logging of the (pre-binding) `SourceFile`, then a `post-binding:` debug-logging of the `SourceFile` where the `SourceFile` node's `locals` is a symbol table mapping from the `__String` `"a"` to a `Symbol` with `escaped_name: "a"` and `declarations` a `Vec` containing the `VariableDeclaration` AST node that declared `a`, and the `symbol` on the `BaseNode` inside that `VariableDeclaration` should be a populated weak reference (to the `Symbol`)  and then it should panic during type-checking

Based on `variable-declaration`
